genestack_client/genestack_connection.py

```diff
@@ -95,8 +95,14 @@ def check_version(self):
         :param version: version in format suitable for distutils.version.StrictVersion
         :return: None
         """
-        compatible_version = self.application('genestack/clientVersion'
-                                             ).invoke('getCompatibleVersion')
+        try:
+            compatible_version = self.application('genestack/clientVersion'
+                                                 ).invoke('getCompatibleVersion')
+        except GenestackServerException, e:
+            # We don't know what happened, but it might be due to incompatible client/API versions.
+            # Rethrow exception, making sure we tell the user to update.
+            e.prompt_update = True
+            raise e
```

Comment:
We can simply throw here a `GenestackVersionException` with corresponding message.

Currently message is hardcoded inside `GenestackVersionException`. We can change this logic and provide the message to constructor.